app/support/reservations/validations.rb

```diff
@@ -34,6 +35,12 @@ def starts_before_ends
     end
   end
 
+  def starts_before_cutoff
+    if product.cutoff_time && reserve_start_at && in_the_future? && reserve_start_at_changed?
+      errors.add("reserve_start_at", "must be before the cutoff time for this instrument") if reserve_start_at < Time.now + product.cutoff_time.hours
```

Comment:
This file hasn't been localizing messages generally, but it probably should. Something like `errors.add(:reserve_start_at, :after_cutoff)`?